### PR TITLE
feat: add supportsInterface

### LINKED ISSUE #205: Add `.supportsInterface()` method
```js
ERC725.supportsInterface(interfaceId or interfaceName, options) > bool
```

```js
myErc725.supportsInterface(interfaceId or interfaceName) > bool
```

interfaceName [from our [docs](https://docs.lukso.tech/standards/smart-contracts/interface-ids)]:
```
ERC1271,
ERC725X,
ERC725Y,
LSP0ERC725Account,
LSP1UniversalReceiver,
LSP1UniversalReceiverDelegate
LSP6KeyManager,
LSP7DigitalAsset,
LSP8IdentifiableDigitalAsset,
LSP9Vault,
```

`options` :
```
{
  address: '0x23456789',
  provider: 'https://....'
}
```

This object can also be reused for other functions on the class, such as:

```js
ERC725.getOwner({
address: '0x23456789',
provider: 'https://....'
}) > address
```

 // create new issue

On instance, `address` is optional -> defaults to the address used on the instance.

Returns

## Reference:

- https://docs.lukso.tech/standards/smart-contracts/interface-ids

### PR DESCRIPTION
**Pull request by fhildeb**
### What kind of change does this PR introduce (bug fix, feature, docs update, ...)?

This PR introduces interface ID checkups as standalone modular function within the library that can be called directly on the ERC725 object or a custom smart contract address. For convenience, the detection tool can not only handle interface IDs, but also a variety of names.

### What is the current behaviour (you can also link to an open issue here)?

At the moment, interface detection has to be done by calling `supportsInterface()` on the smart contract directly while also having to know the right interface ID.

closes #205 

### What is the new behaviour (if this is a feature change)?

You can use `myERC725.supportsInterface(interfaceIdOrName)` or `ERC725.supportsInterface(interfaceIdOrName, options)` to check if the ERC725 object or smart contract supports a specific interface. When you use the function on your instantiated ERC725 object, it will use its address and provider by default. Otherwise, you need to specify them in the `options` parameter.

### Other information:

The `interfaceName` will only check for the latest version of the standard's `interfaceID`, which can be found in `src/constants/interfaces`. For LSPs, the `interfaceIDs` are taken from the latest release of the [@lukso/lsp-smart-contracts](https://github.com/lukso-network/lsp-smart-contracts) library.

#### Default supported interface names:

``` 
ERC1271,
ERC725X,
ERC725Y,
LSP0ERC725Account,
LSP1UniversalReceiver,
LSP1UniversalReceiverDelegate
LSP6KeyManager,
LSP7DigitalAsset,
LSP8IdentifiableDigitalAsset,
LSP9Vault
```

### CODE REVIEW

File: src/lib/detector.test.ts [hunk 1]
```diff
@@ -22,11 +22,13 @@
...
 import { expect } from 'chai';
 import * as sinon from 'sinon';
+import assert from 'assert';
+
 import { INTERFACE_IDS_0_7_0 } from '../constants/interfaces';
 
 import { supportsInterface } from './detector';
 
-describe('supportsInterface', () => {
+describe.only('supportsInterface', () => {
```
Comment (Hugoo) [inline src/lib/detector.test.ts:31]: Please remove the only

File: src/lib/detector.test.ts [hunk 2]
```diff
@@ -62,4 +64,19 @@ describe('supportsInterface', () => {
...
+
+  it('should throw when wrong rpcUrl was used', () => {
+    const contractAddress = '0xcafecafecafecafecafecafecafecafecafecafe';
+    const interfaceId = INTERFACE_IDS_0_7_0.LSP1UniversalReceiver;
+    const rpcUrl = 'https://rpc.my.wrong.rpc';
+
+    assert.throws(
+      () => {
+        // @ts-ignore
+        // eslint-disable-next-line no-new
```
Comment (Hugoo) [inline src/lib/detector.test.ts:76]: Why ts-ignore?

**Review by Hugoo — CHANGES_REQUESTED** [review]

File: docs/classes/ERC725.md [hunk 3]
```diff
@@ -1178,3 +1178,89 @@ await erc725.isValidSignature(
...
+
+You can use this function if you need to check if the ERC725 object or a smart contract supports a specific interface (by ID or name). When you use the function on your instantiated ERC725 object, it will use its address and provider by default. Otherwise, you need to specify them in the `options` parameter.
+
+:::caution
+The `interfaceId` is not the most secure way to check for a standard, as they could be set manually.
+:::
+
+#### Parameters
+
+##### 1. `interfaceIdOrName` - String of ID or name
```
Comment (Hugoo) [inline docs/classes/ERC725.md:1200]:
```suggestion
##### 1. `interfaceIdOrName` - String
```

File: docs/classes/ERC725.md [hunk 4]
```diff
@@ -1178,3 +1178,89 @@ await erc725.isValidSignature(
...
+| `LSP8IdentifiableDigitalAsset`  | [LSP-8: Identifiable Digital Asset](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-8-IdentifiableDigitalAsset.md) |
+| `LSP9Vault`                     | [LSP-9: Vault](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-9-Vault.md)                                         |
+
+:::info
+
+The **`interfaceName`** will only check for the latest version of the standard's **`interfaceID`**, which can be found in `src/constants/interfaces`. For LSPs, the **`interfaceIDs`** are taken from the latest release of the [@lukso/lsp-smart-contracts](https://github.com/lukso-network/lsp-smart-contracts) library.
+
+:::info
+
+##### 2. `options` - Object of contract address and RPC URL
```
Comment (Hugoo) [inline docs/classes/ERC725.md:1223]:
```suggestion
##### 2. `options` - Object
```

**Review by Hugoo — APPROVED** [review]

**Comment by fhildeb** [comment]
Looks good, thanks!

**Pull request merged**